tracer/tracer.go

```diff
@@ -1186,3 +1186,39 @@ func (t *Tracer) AttachUProbes(uprobes []string) error {
 func (t *Tracer) HandleTrace(bpfTrace *host.Trace) {
 	t.processManager.HandleTrace(bpfTrace)
 }
+
+// UpdateSamplingFrequency dynamically updates the sampling frequency for all perf events.
+// Internally converts frequency (Hz) to period (nanoseconds) and calls UpdatePeriod on each event.
+func (t *Tracer) UpdateSamplingFrequency(newSamplesPerSecond int) error {
+	if newSamplesPerSecond <= 0 {
+		return fmt.Errorf("invalid sampling frequency: %d", newSamplesPerSecond)
+	}
+
+	events := t.perfEntrypoints.WLock()
+	defer t.perfEntrypoints.WUnlock(&events)
+
+	if len(*events) == 0 {
+		return errors.New("no perf events available to update")
+	}
+
+	// Calculate the new sampling period in nanoseconds
+	// period = 1 second / frequency
+	period := uint64(time.Second.Nanoseconds() / int64(newSamplesPerSecond))
+
+	var updateErrors []error
+	for i, event := range *events {
+		if err := event.UpdatePeriod(period); err != nil {
```

Comment:
This doesn't change the size of the [per-CPU buffers](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/b83940a7c04076dedeb3ef37415addba52a991bc/tracer/events.go#L139-L140) where the events are written, so we can expect to see lots of dropped events with higher sampling frequencies and full CPU utilization.

Reply:
It seems that considering the overall situation, it is not appropriate to use the event method directly, right?

Reply:
One way is to introduce a `maxSamplingFrequency` configuration option and size the per-CPU buffers by it (worst-case). Then in the `UpdateSamplingFrequency` method you would not allow the frequency to exceed the configured max. 

Keep in mind though that the profiler isn't currently designed with very high (e.g. in the thousands) sampling frequencies in mind and there will be unexpected issues when that's the case.